### PR TITLE
Implement course deletion

### PR DESCRIPTION
**Pull request by maexled**
Closes Gamify-IT/issues#214

I am a bit lost in designing. Please leave feedback if you have suggestions.

### CODE REVIEW

**Comment by robojones** [comment]
When deletion successful, include this for additional festiveness:

<https://www.npmjs.com/package/vue-confetti>

**Comment by delvh** [comment]
As much as I like meming, I veto against showing users that they are better of without using our platform.

**Comment by maexled** [comment]
> When deletion successful, include this for additional festiveness:
> 
> https://www.npmjs.com/package/vue-confetti

With feedback i meant more like where to place the delete button. But I think this task can be handled in Gamify-IT/issues#217

**Comment by maexled** [comment]
Review the code please, I want to merge soon

**Review by Aaron2000119 — APPROVED** [review]

**Comment by cdingm** [comment]
image
when i create a new course, and then delete it i get an error. But when back on the overview the course is now missing.

if i try to delete one of the dummy courses it also fails, but the course stays on the overview

**Comment by maexled** [comment]
> image when i create a new course, and then delete it i get an error. But when back on the overview the course is now missing.
> 
> if i try to delete dummy course with ID 15 it also fails, but the course stays on the overview the other dummys can be deleted but throw the error that they couldnt be deleted
> 
> this is the error if it helps: image

This is fixed in Gamify-IT/overworld-backend#26. This PR is just for frontend functionality, if you want to ensure that it works, you need to use the version from the PR or wait till the bug is merged.

**Review by cdingm — APPROVED** [review]
tested with the fix of https://github.com/Gamify-IT/overworld-backend/pull/26
works for me

File: src/views/CourseView.vue [hunk 1]
```diff
@@ -112,47 +113,63 @@ function toggleCourseSwitch() {
     }
   });
 }
+
+const showDeleteConfirmation = ref(false);
```
Comment (delvh) [inline src/views/CourseView.vue:117]: Unnecessary.
This variable has no purpose, the mechanism also works without it.

File: src/views/CourseView.vue [hunk 2]
```diff
@@ -112,47 +113,63 @@ function toggleCourseSwitch() {
...
+
+function confirmDeleteCourse() {
```
Comment (delvh) [inline src/views/CourseView.vue:119]: This method should perhaps be renamed to `deleteCurrentCourse`

File: src/views/CourseView.vue [hunk 3]
```diff
@@ -169,4 +186,24 @@ function toggleCourseSwitch() {
...
+  >
+    Are you sure you want to delete course <b>{{ course.courseName }}</b> of
+    semester {{ course.semester }}?
+    <template v-slot:footer>
+      <b-button
+        variant="outline-secondary"
+        @click="showDeleteConfirmation = false"
+      >
+        Cancel
+      </b-button>
```
Comment (delvh) [inline src/views/CourseView.vue:203]: You don't even need the full footer, the `v-slot:ok` should suffice.

Comment (maexled) [reply src/views/CourseView.vue:203]: Like this?
grafik
This is not working.
Also, I checked the source code of this component here (https://github.com/cdmoro/bootstrap-vue-3/blob/main/packages/bootstrap-vue-3/src/components/BModal.vue) and I can not see another template slot.

Comment (delvh) [reply src/views/CourseView.vue]: Ah, could it be that bootstrap-vue-3 does not support that yet?
I mean, it's a rewrite of the library, barely a year old.
Could be that that isn't supported yet.
I simply looked inside the docs of the original bootstrap-vue.

Comment (maexled) [reply src/views/CourseView.vue]: I found now a better solution, check [change button variant and text by variables, not by own template](https://github.com/Gamify-IT/lecturer-interface/pull/15/commits/067f792a2c92633e9d411789bc81a1209b28dc88)

File: src/views/CourseView.vue (hunk 1 above)
Comment (delvh) [reply src/views/CourseView.vue:117]: Yes, but if I'm not mistaken, not passing a `v-model` is allowed in vue.
And if that's the case, then it is unnecessary as you can simply ignore it and only do something (deleting the course) when the delete button of the modal has been clicked.

Comment (maexled) [reply src/views/CourseView.vue]: Ok, I think now I got it here [remove v-model of delete modal](https://github.com/Gamify-IT/lecturer-interface/pull/15/commits/7d043456630891bd7d9e292d722f7ac0c9696851)

**Review by delvh — APPROVED** [review]

**Pull request merged**